src/ESA.cpp

```diff
@@ -0,0 +1,232 @@
+    if (verbose) cout << "Done" << endl;
+    return 0;
+  }
+  catch (const runtime_error& err) {
+    cerr << err.what() << endl;
+    return 1;
+  }
+}
+
+
+// ESA main function
+void ESA::doESA() {
+
+// Start timing
+double tic = omp_get_wtime();
+// Compute ESA static local field correction
+computeSlfc();
+// Compute static structure factor
+computeSsf();
+// End timing
+double toc = omp_get_wtime();
+// Print diagnostic
+if (verbose) {
+    printf("Elapsed time: %f seconds\n", toc - tic);
+    fflush(stdout);
+}
+  
+}
+
+
+void ESA::computeSlfc() {
+
+    // Get the degeneracy and coupling parameters
+    const double theta = in.getDegeneracy();
+    const double rs = in.getCoupling();
+
+    // Constants for the computation of the static local field correction
+    const double g0aa1 = 18.4376509088802;
+    const double g0ba1 = 24.1338558554951;
+    const double g0ba2 = 1.86499223116244;
+    const double g0a0 = 0.18315;
+    const double g0ab1 = -0.243679875065179;
+    const double g0bb1 = 0.252577;
+    const double g0bb2 = 0.12704315679703;
+    const double g0b0 = -0.0784043;
+    const double g0ac1 = 2.23662699250646;
+    const double g0ac2 = 0.448936594134834;
+    const double g0bc1 = 0.445525554738623;
+    const double g0bc2 = 0.408503601408896;
+    const double g0c0 = 1.02232;
+    const double g0ad1 = 0.0589015402409623;
+    const double g0bd1 = -0.598507865444083;
+    const double g0bd2 = 0.513161640681146;
+    const double g0d0 = 0.0837741;
+
+    const double Eta = 3;
+    const double Ax = 2.64;
+    const double Bx = 0.31;
+    const double Cx = 0.08;
+    const double aa1 = 0.66477593;
+    const double aa2 = -4.59280227;
+    const double aa3 = 1.24649624;
+    const double ba1 = -1.27089927;
+    const double ba2 = 1.26706839;
+    const double ba3 = -0.4327608;
+    const double ca1 = 2.09717766;
+    const double ca2 = 1.15424724;
+    const double ca3 = -0.65356955;
+    const double ab1 = -1.0206202;
+    const double ab2 = 5.16041218;
+    const double ab3 = -0.23880981;
+    const double bb1 = 1.07356921;
+    const double bb2 = -1.67311761;
+    const double bb3 = 0.58928105;
+    const double cb1 = 0.8469662;
+    const double cb2 = 1.54029035;
+    const double cb3 = -0.71145445;
+    const double ac1 = -2.31252076;
+    const double ac2 = 5.83181391;
+    const double ac3 = 2.29489749;
+    const double bc1 = 1.76614589;
+    const double bc2 = -0.09710839;
+    const double bc3 = -0.33180686;
+    const double cc1 = 0.56560236;
+    const double cc2 = 1.10948188;
+    const double cc3 = -0.43213648;
+    const double ad1 = 1.3742155;
+    const double ad2 = -4.01393906;
+    const double ad3 = -1.65187145;
+    const double bd1 = -1.75381153;
+    const double bd2 = -1.17022854;
+    const double bd3 = 0.76772906;
+    const double cd1 = 0.63867766;
+    const double cd2 = 1.07863273;
+    const double cd3 = -0.35630091;
+
+    double aa = aa1 + aa2 * theta + aa3 * pow(theta, 3.0/2.0);
```

Comment:
All the doubles that you define between here and line 153 declare them as const double